### PR TITLE
validators: Date validator

### PR DESCRIPTION
**Pull request by JakubNvk**
* added date validator
* replaced DateField with StringField for date_of_birth

Fixes: #8 

Signed-off-by: JakubNvk <jakub.novak@gmail.com>

### CODE REVIEW

File: eprihlaska/forms.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def process_formdata(self, valuelist):
 class MoreDetailPersonalDataForm(FlaskForm):
     birth_no = StringField(label=c.BIRTH_NO,
                            validators=[BirthNoValidator()])
-    date_of_birth = ePrihlaskaDateField(label=c.BIRTH_DATE,
-                                        format='%d.%m.%Y',
-                                        default=DEFAULT_DATE)
+    date_of_birth = StringField(label=c.BIRTH_DATE,
+                                validators=[DateValidator()],
+                                default='DD.MM.RRRR')
```
Comment (mrshu) [inline eprihlaska/forms.py]: @JakubNvk Co by sme tu presne potrebovali je placeholder, ale to zial asi `StringField` neznesie...

Asi by som to odporucal sem nedat.

File: eprihlaska/forms.py [hunk 2]
```diff
@@ -76,6 +76,8 @@ class MoreDetailPersonalDataForm(FlaskForm):
 
     place_of_birth = SelectField(label=c.BIRTH_PLACE,
                                  choices=choices_from_csv(DIR + '/data/obce.csv',
+                                                          ['id', 'Názov obce'],
```
Comment (mrshu) [inline eprihlaska/forms.py]: @JakubNvk Neviem presne, ako sa to sem dostalo, ale toto sem dva krat netreba. Malo by to tiez riesit problem s `TypeError: choices_from_csv() got multiple values for argument 'fmt'`

File: eprihlaska/forms.py [hunk 3]
```diff
@@ -221,8 +223,6 @@ class FurtherStudyInfoForm(FlaskForm):
                                     validators=[validators.Optional()])
 
     will_take_external_mat_matura = BooleanField(label=c.WILL_TAKE_EXT_MAT)
-    will_take_scio = BooleanField(label=c.WILL_TAKE_SCIO)
```
Comment (mrshu) [inline eprihlaska/forms.py]: @JakubNvk Toto bolo zrejme typo -- myslim, ze toto by malo zostat.

File: eprihlaska/validators.py [hunk 4]
```diff
@@ -32,3 +32,21 @@ def __call__(self, form, field):
...
+class DateValidator:
+    def __init__(self, date=None, message=None):
+        self.date = date
+
+        if not message:
+            message = 'Správny formát je DD.MM.RRRR'
+        self.message = message
+
+    def __call__(self, form, field):
+        date_reg = re.compile(r'^(?:(?:31(\.)(?:0?[13578]|1[02]))\1|(?:(?:29|30)(\.)(?:0?[1,3-9]|1[0-2])\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:29(\.)(?:0?2)\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\d|2[0-8])(\.)(?:(?:0?[1-9])|(?:1[0-2]))\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$')
```
Comment (mrshu) [inline eprihlaska/validators.py]: @JakubNvk Toto je hrozne komplikovany regex. Slo by ho trochu zjednodusit? 

Myslim, ze by nam uplne stacilo `r'^(\d\d)\.(\d\d)\.(\d\d\d\d)$'` a potom vyskusat, ci je ten datum validny, napriklad nejak [takto](https://stackoverflow.com/questions/16870663/how-do-i-validate-a-date-string-format-in-python). Ci sa to takto zda byt lepsie?

Este by som odporucal tento kompilovany regex dat do `__init__`-u, aby sa to nekompilovalo znova a znova.

Vdaka!

File: eprihlaska/forms.py (hunk 1 above)
Comment (JakubNvk) [reply eprihlaska/forms.py]: Trosku sa bojim, ze tak polovica pouzivatelov zada zly format (zly podla nasich standardov) a budu to musiet menit, ak to tam nenechame.

File: eprihlaska/validators.py (hunk 4 above)
Comment (JakubNvk) [reply eprihlaska/validators.py]: Cokolvek je lepsie ako henta abominacia...prerobim to :)

File: eprihlaska/forms.py (hunk 2 above)
Comment (JakubNvk) [reply eprihlaska/forms.py]: Uprimne netusim, spadol mi pocas prace terminal - je dost mozne ze to narobilo zvlastny bordel..

**Comment by mrshu** [comment]
Vdaka @JakubNvk!

**Pull request merged**